### DISCUSSION TITLE
hi, I wrote my first code, if you are interested, you can rate and point out the errors

### DISCUSSION THREAD

**Discussion by timur-blip** [community/community#78551] [Programming Help]
### Select Topic Area

Product Feedback

### Body

#include <iostream>
using namespace std;
void enter_converted(unsigned int arr[]);
int cout_converted(unsigned int n, unsigned int base);
int main()
{
    unsigned n, base;
    cout << "\t" << "Converting a number from decimal to any number system\n\t Enter  decimal number\n ";
    cout << "\t";cin >> n;
    cout << "\t" << "Enter the base\n";
    cout << "\t"; cin >> base;
    cout << "\t";cout_converted(n, base);
    cout << "\n\n\n\n\n\n";
    return 0;
}
void enter_converted(unsigned int arr[])
{
    for(int k = 31;k>0;k--)
    {
        if (arr[k] == 0)
            continue;
        else
        {
            for (;k >= 0;k--)
            {
                cout << arr[k];
            }
        }
    }
}
int cout_converted(unsigned int n, unsigned int base)
{
    unsigned int dig, i = 0, arr[32]{0};
    do
    {
        arr[i] = n % base;
        dig = n % base;
        n = n / base;
        i++;
    } while (n>1);
    dig = n % 2;
    if (dig != 0)
    {
        arr[i + 1] = dig;
    }
    enter_converted(arr);
    return 1;
}

**Comment by timur-blip** [comment]
base < 10

**Comment by erickdc7** [comment]
Congratulations on writing your first code! Here's some feedback on your code:

```cpp
#include <iostream> // Include the correct header file for I/O operations
using namespace std;

void enter_converted(unsigned int arr[]);
int cout_converted(unsigned int n, unsigned int base);

int main() {
    unsigned int n, base; // Use 'unsigned int' for consistency
    cout << "\tConverting a number from decimal to any number system\n\tEnter decimal number\n\t";
    cin >> n;
    cout << "\tEnter the base\n\t";
    cin >> base;
    cout_converted(n, base); // Removed 'cout <<' as the function already handles output
    cout << "\n\n\n\n\n\n";
    return 0;
}

void enter_converted(unsigned int arr[]) {
    for (int k = 31; k >= 0; k--) { // Changed 'k > 0' to 'k >= 0' to include arr[0]
        if (arr[k] != 0) { // Simplified the logic by removing unnecessary 'continue'
            for (; k >= 0; k--) {
                cout << arr[k];
            }
            break; // Added 'break' to exit the loop after printing
        }
    }
}

int cout_converted(unsigned int n, unsigned int base) {
    unsigned int i = 0, arr[32] = {0};
    while (n > 0) { // Changed to 'while' loop for clarity
        arr[i++] = n % base;
        n /= base; // Simplified division
    }
    enter_converted(arr);
    return 1; // Consider changing the return type to 'void' if '1' is not used
}
```

**Key Points:**
- **Include Correct Header**: Make sure to include the correct header file for input/output operations.
- **Loop Conditions**: Pay attention to loop conditions to ensure all elements are processed.
- **Function Return Type**: Consider the return type of functions and whether it's necessary.
- **Code Simplification**: Simplify the code where possible to improve readability and efficiency.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @timur-blip ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
